### ISSUE TITLE
[Feature] Add `merchant` and `date` fields to manual requests

### ISSUE THREAD

**Issue by Beamanator** [Expensify/App#23245]
Currently we only show "To", "Amount", and "Description" fields for each manual request. We're in the process of also gathering "merchant" & "date" info for each request, so let's show that in the manual request confirm page:

<img width="415" alt="Screenshot 2023-07-20 at 10 19 31 AM" src="https://github.com/Expensify/App/assets/3885503/53529cf8-3c9f-411a-8b27-81898f625408">

In the near future, we'll have two new properties in `transaction` data:
- `merchant` - the original merchant data, cannot be modified
- `modifiedMerchant` - if the user modifies the merchant field, we'll set that data here

1. We want to show these new fields behind a “Show more” button that will toggle a new state, showAllFields.
2. When set to true, all fields will be displayed
    - For merchant data, we'll show the `modifiedMerchant` data if it exists, otherwise we'll default to `merchant`
3. When set to false, we'll show the "Show more" button with a horizontal line as in the screenshot
4. Note: For now, merchant & date fields will not be editable so we do not need the `>` like other push-to-page fields

Note: Needs this Auth PR to get to prod so we get merchant data in onyx transaction key: https://github.com/Expensify/Auth/pull/8325

**Comment by lukemorawski** [comment]
Hey. It's Lucas from Callstack. Happy to work on that feature!

**Comment by Beamanator** [comment]
Thanks @lukemorawski !

**Comment by Beamanator** [comment]
Hey @lukemorawski as this is a high priority for us I've switched the priority to `Daily` and I'd highly appreciate daily updates on your progress iff possible 🙏

**Comment by Beamanator** [comment]
Oh and by the way @lukemorawski the Auth PR changes are now in production!

**Comment by lukemorawski** [comment]
@Beamanator Hey Alex. Got the UI ready, but need a hint on plugin in to the real data source. Where should I look for those `merchant`, `modifiedMerchant` and `date` fields in Onyx?

**Comment by Beamanator** [comment]
Awesome @lukemorawski ! `merchant` and `modifiedMerchant` will show up in the `transactions` onyx collection, and you'll only get that in money request reports - @MariaHCD @Gonals I believe that should work and it's live, right?

**Comment by lukemorawski** [comment]
@Beamanator yeah, thanks I can see that collection in Onyx and those new fields there. The problem is I can't see the a new transaction there upon creating a new money request. Perhaps I'm doing something wrong? Could please provide the user path to that screen? (sorry, I'm still a bit new to the app and it's data flow)

**Comment by Beamanator** [comment]
Yeah good question @lukemorawski ! So you don't see the `transaction` data in Onyx? Did you try Request Money specifically? I just tested that in staging and it works well, I can see `transactions_<id>` in Onyx

**Comment by lukemorawski** [comment]
No, I can see them. It's just when I'm creating a new money request, that new transaction is not yet stored in Onyx. That's why I can't access it and display those new fields.

**Comment by Beamanator** [comment]
Sorry not sure who's not being clear 😅 

> No, I can see them

So you do or don't see some transaction data in Onyx? 😅

**Comment by lukemorawski** [comment]
The proposal is in callstack internal review

**Comment by Beamanator** [comment]
Thanks for the update @lukemorawski - would you mind letting us know when you estimate a PR will be up for this? 🙏

**Comment by lukemorawski** [comment]
@Beamanator should be ready today

**Comment by lukemorawski** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
We want to add new fields to the money request page - `merchant` and `date`.
Those new fields need to be hidden under the “Show more” button.
Values of those fields are taken from the “iou” object created when creating a new money request.
[@reviewers](https://callstack-hq.slack.com/admin/user_groups)
### What changes do you think we should make in order to solve the problem?
We need to add a simple bool state flag to the `MoneyRequestConfirmationList.js` and a small button with horizontal line that will toggle this flag. The flag will control the visibility of two new `MenuItemWithTopDescription` components.
New props would need to be introduced to the `MoneyRequestConfirmationList` component - one for passing the date and the other to pass the merchant value.
Those props are passed from the `MoneyRequestConfirmPage` component, and are taken from an Onyx subscription to ‘iou’ object
In order get the new default values in the ‘iou’ object we need to add them to the object merged in `resetMoneyRequestInfo` function from IOU.js. Default value for merchant is an empty string and date is current date.
In future PRs those values will be editable.
### What alternative solutions did you explore? (Optional)
None.

https://github.com/Expensify/App/assets/134388952/a5986175-4ddd-48f0-9428-3cb95bce2506

**Comment by Beamanator** [comment]
@lukemorawski so far that's looking great!

One question, more for @MariaHCD and @Gonals and @shawnborton - when a user clicks the "Show more" button, I think we want it to disappear so they can't re-hide the other fields (merchant & date), right?

**Comment by Beamanator** [comment]
Amazing, thanks @lukemorawski ! I think you're good to raise the PR whenever you can, as that tiny change we discussed above should be quick to implement 👍 🙏

**Comment by dylanexpensify** [comment]
Wooo! We're getting close to finishing this one, nice work! Now let's blast past that finish line! 🏁 💨

**Comment by Beamanator** [comment]
Adding @situchan since they reviewed the PR 👍 

@CortneyOfstad I'm not sure how the automation didn't work for this, but [the PR](https://github.com/Expensify/App/pull/23648) was deployed to production two weeks ago 😅  So we're just waiting for payment here 👍

**Comment by dylanexpensify** [comment]
I'll take this on for Cortney! Payment coming

**Comment by situchan** [comment]
> @situchan confirm for me if you're paid in NewDot?

I am using upwork

**Comment by dylanexpensify** [comment]
Offer sent!

**Comment by dylanexpensify** [comment]
payment sent, contract closed! ty!
